kod/object/passive/ornobj.kod

```diff
@@ -735,14 +749,143 @@ messages:
    {
       local f;
 
-      f = LOOK_NO;
+      f = 0;
+      if NOT pbLook { f = f | LOOK_NO; }
       if pbHanging { f = f | OF_HANGING; }
       if pbFlicker { f = f | FLICKERING_YES; }
       if pbNoMoveOn { f = f | MOVEON_NO; }
       
       return f;
    }
 
+   ShowDesc()
+   {
+      if psCustomDesc <> $
+      {
+         AddPacket(4, oo_custom_template_rsc, 4, psCustomDesc);
+
+         return;
+      }
+
+      propagate;
+   }
+
+   StartDecayTimer(d=0, h=0, m=0, s=0)
+   "Admin supported\n"
+   "Starts a timer for this object which will automatically delete it upon expiring.\n"
+   "Can specify the time using h for hours, m for minutes, and/or s for seconds."
+   {
+      local iTime;
+
+      iTime = d * 86400000 + h * 3600000 + m * 60000 + s * 1000;
```

Comment:
Easier to understand: 1000 * (s + 60 * (m + 60 * (h + 24 * d))).

I'm a bit concerned about rollover here.  2^28 / 86400000 = 3.1, i.e. if you set days to >= 4, the Blakod int value will rollover, and you won't get the time you want.  Should add a check for this.  And it might be only 2^27 depending how signed values work.. please test these large values.

Reply:
Done, although I reversed the order to be days ... -> ... seconds for even more clarity.  I'd assumed (big mistake, I don't usually assume) timers could handler large numbers since raw time values can be quite big, but it seems things go south at 25+ days, so I set the cutoff to only allow 20 or fewer.  It's still a very rudimentary check since one could potentially do 20 days, then like 5000 hours and still cause the rollover.

Overall I'm a bit confused about something with this.  When I debug-output iTime after passing 25 days, it shows a negative value, however the iTime > 0 checks still pass?

Reply:
Raw time values are stored as 64-bits in the server, but Blakod values are 28 bits.  (Or at least, they were until recently when I made some server changes.  Did you clean and rebuild the server recently?)  And I see that 2^28 / 86400000 = 3.1, so I think we don't understand exactly what's happening here.  Putting a cutoff at a random 20 days doesn't seem safe until we do.

Reply:
Yeah the server's been rebuilt fairly recently and I keep my repo synced, so everything should be up to date.  Things got weirder with this too--either I was losing my mind and tired, or it seemed that, when I skipped assigning the result to the local variable and instead put the equation directly into CreateTimer(), the days could go to 40 and possibly a bit higher?  I checked this with a Debug(GetTimeRemaining(timerHere)) after creating it and the output was an ever-increasing value all the way to 40, then I skipped to 50 and it failed.

Overall I don't feel confident in how this is behaving (results vs expectations) and I'm also not convinced this particular feature is even worth the trouble.  So it's nuked.